script/tests/anchor.blocks.spec.ts

```diff
@@ -0,0 +1,119 @@
+import * as util from "util";
+import * as child_process from "child_process";
+const exec = util.promisify(child_process.exec);
+let interval,
+  triggerCount = 0,
+  foundMicroBlock = false;
+
+import { getAccountDetail, getMicroBlock, getAnchorBlock } from "../utils/curl";
+import { wait } from "../utils/helper";
+
+const runAnchorBlockTest = async (options: any) => {
+  triggerCount++;
+  console.log(`Trigger script count: ${triggerCount}`);
+
+  const { accountId, recipient } = options;
+  try {
+    // step 1: Fetch nonce from account id
+    console.log("Fetching Nonce...");
+    const nonce = await fetchNonce(accountId);
+    console.log("Nonce fetched", nonce);
+
+    // step 2: Run the stx command for send_tokens to fetch the TXID
+    console.log("Sending tokens and fetching TXID...");
+    const txid = await sendTokens(nonce, recipient);
+    console.log("TXID fetched", txid);
+
+    // step 3: CURL the microblock API to make sure that TXID exist
+    if (!foundMicroBlock) {
+      await wait(10000);
+      console.log("Calling microblock API...");
+      const isTXIDExistMB = await checkTXIDInMicroBlock(txid);
+      if (isTXIDExistMB) {
+        foundMicroBlock = true;
+        console.log(`The transaction ${txid} is present in Micro Block API`);
+      } else {
+        console.log(
+          `The transaction ${txid} is not present in Micro Block API`
+        );
+      }
+    }
+
+    // step 4: CURL the Anchor block API to make sure that TXID exist
+    console.log("Calling Anchor block API...");
+    const isTXIDExistAB = await checkTXIDInAnchorBlock(txid);
+    if (isTXIDExistAB) {
+      console.log(`The transaction ${txid} is present in Anchor block API`);
+      process.exit(0);
+    } else {
+      console.log(`The transaction ${txid} is not present in Anchor block API`);
+    }
+    if (triggerCount === 5) {
+      console.log("Time limit reached: 30 minutes");
+      clearInterval(interval);
+      process.exit(0);
+    }
+  } catch (e) {
+    console.error(
+      "An error occurred while processing the anchor block script",
+      e
+    );
+    clearInterval(interval);
+    process.exit(1);
+  }
+};
+
+export function triggerAnchorBlockTests(options: any) {
+  runAnchorBlockTest(options);
+
+  interval = setInterval(() => {
+    runAnchorBlockTest(options);
+  }, 6 * 60 * 1000); // every 6 minutes
+}
+
+async function fetchNonce(accountId: string): Promise<number> {
+  const accountDetail = await getAccountDetail(accountId);
+  if (!accountDetail) {
+    console.log(`No account detail found for account id - ${accountId}`);
+    return 0;
+  }
+  return accountDetail.nonce;
+}
+
+async function sendTokens(nonce: number, recipient: string): Promise<any> {
+  const { stdout, stderr } = await exec(
+    `stx -t send_tokens ${recipient} 12345 999 ${nonce} fd3609e8b9352facf360c950d362afe8f0f108b9041750f54b017efc479efeb001`
```

Comment:
I'm not too familiar with the stx cli... is any of these inputs sensitive info? In that case you could save it as a Github actions secret in the repo and use it as an input via an ENV var

Reply:
Sure I can do that this is a test account so I just threw the private key into the script but I do see your point.